frontend_tests/node_tests/message_list_view.js

```diff
@@ -99,47 +99,58 @@ test("msg_moved_var", () => {
 
     (function test_msg_moved_var() {
         const messages = [
-            // no edits: Not moved.
-            build_message_context(),
-            // stream changed: Move
+            // stream changed: MOVED
             build_message_context({
-                edit_history: [{prev_stream: "test_stream", timestamp: 1000, user_id: 1}],
+                edit_history: [{prev_stream: 1, timestamp: 1000, user_id: 1}],
             }),
-            // topic changed: Move
+            // topic changed - not resolved / unresolved: MOVED
             build_message_context({
-                edit_history: [{prev_topic: "test_topic", timestamp: 1000, user_id: 1}],
+                edit_history: [
+                    {prev_topic: "test_topic", topic: "new_topic", timestamp: 1000, user_id: 1},
+                ],
             }),
-            // content edited: Edit
+            // content edited: EDITIED
             build_message_context({
                 edit_history: [{prev_content: "test_content", timestamp: 1000, user_id: 1}],
             }),
-            // stream and topic edited: Move
+            // stream and topic edited: MOVED
             build_message_context({
                 edit_history: [
-                    {prev_stream: "test_stream", timestamp: 1000, user_id: 1},
-                    {prev_topic: "test_topic", timestamp: 1000, user_id: 1},
+                    {prev_stream: 1, timestamp: 1000, user_id: 1},
+                    {prev_topic: "test_topic", topic: "new_topic", timestamp: 1000, user_id: 1},
                 ],
             }),
-            // topic and content changed: Edit
+            // topic and content changed: EDITED
             build_message_context({
                 edit_history: [
-                    {prev_topic: "test_topic", timestamp: 1000, user_id: 1},
+                    {prev_topic: "test_topic", topic: "new_topic", timestamp: 1000, user_id: 1},
                     {prev_content: "test_content", timestamp: 1001, user_id: 1},
                 ],
             }),
-            // stream and content changed: Edit
+            // topic resolved and content changed: EDITED
+            // added content change to get the container field
             build_message_context({
                 edit_history: [
-                    {prev_content: "test_content", timestamp: 1000, user_id: 1},
-                    {prev_stream: "test_stream", timestamp: 1001, user_id: 1},
+                    {
+                        prev_topic: "test_topic",
+                        topic: "✔ test_topic",
+                        prev_content: "test_content",
+                        timestamp: 1000,
+                        user_id: 1,
+                    },
                 ],
             }),
-            // topic, stream, and content changed: Edit
+            // topic unresolved and content changed: EDITED
+            // added content change to get the container field
```

Comment:
I guess we need to add a test case for "only resolved", right?

Reply:
Yes, I expanded the tests in the updated commit so that all the single edit possibilities are tested as well as two multiple edit history entries. There is one test - for a topic unresolved - that should never happen because it doesn't also have an associated topic resolved log, but I thought we should just test the logic of the loop just in case ...

Reply:
Yep, I think this is the right testing strategy.